docs/snips_structure.md

```diff
@@ -0,0 +1,231 @@
+
+##Template Content Structure
+
+When you use Snips, you should specify template content and its content structure is fixed as well.
+
+You should manage template content as follow: 
+
+``` sh
+└──template
+      └──manifest.json
+      └──service.tmpl
+      └──shared.tmpl
+      └──sub_service.tmpl
+      └──types.tmpl
+```
+##Introduction of Template Files' Functions
+
+###manifest.json
+
+`manifest.json` is used to specify output files' naming style, extension and tmpl files' name and file path.
+ 
+ You may can manage `manifest.json` as follow:
+ 
+``` Json
+{
+  "output": {
+    "file_naming": {
+      "style": "camel_case",
+      "extension": ".go"
+    }
+  },
+  "template_files": {
+    "service": {
+      "file_path": "service.tmpl",
+      "output_file_naming": {
+        "prefix": "",
+        "suffix": ""
+      }
+    },
+    "sub_service": {
+      "file_path": "sub_services.tmpl",
+      "output_file_naming": {
+        "prefix": "",
+        "suffix": ""
+      }
+    },
+    "types": {
+      "file_path": "types.tmpl",
+      "output_file_naming": {
+        "prefix": "",
+        "suffix": ""
+      }
+    }
+  }
+}
+```
+###service.tmpl
+
+`service.tmpl` is used to generate the code file  towards to service operations.
+
+As an example, `service.tmpl` generated `qingstor` code file of QingStor SDK. (e.g. [qingstor.go](https://github.com/yunify/qingstor-sdk-go/blob/master/service/qingstor.go))
+
+In this template, you should specify the template to generate follow codes:
+
+- Initialization of service.
+- Functions pointing to service.
+
+###shared.tmpl
+
+`shared.tmpl` should define shared functions which other template files need.
+
+###sub_service.tmpl
+
+`sub_service.tmpl`is used to generate the code file  towards to sub service operations.
+
+As an example, `sub_service.tmpl` generated `bucket` code file of QingStor SDK. ([bucket.go](https://github.com/yunify/qingstor-sdk-go/blob/master/service/bucket.go))
+
+In this template,you should specify the template to generate follow codes:
+
+- Initalization of sub service.
+- Functions pointing to sub service.
+
+###types.tmpl
+
+`types.tmpl` is used to generate types code files.
+
+As an example, `types.tmpl` generated `types` code file of QingStor SDK. ([types.go](https://github.com/yunify/qingstor-sdk-go/tree/master/service/types.go)).
+
+In this template, you should specify the template to generate follow codes:
+
+- All types functions.
+
+##Introduction of Snips Intermediate Variables
+
+The intermediate variables are used to store data loading from spec files,and then they will be used to generate code files according to the template files.
+
+###Intermediate Variables Structure
+
+>`Data`
+ >>`Service` `SubService`
+ >>> `Operation`
+ >>>> `Request` `Response`
+ >>>>> `Property`  `StatusCode`
+
+###Introduction of Parameters In Structures
+
+| Structure Name | Element|Element Type | 
```

Comment:
Use <http://www.tablesgenerator.com/markdown_tables> to generate a better table.